functions-and-operators/bit-functions-and-operators.md

```diff
@@ -12,13 +12,288 @@ TiDB supports all of the [bit functions and operators](https://dev.mysql.com/doc
+| 1111000011110000                                                                 |
++----------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+## [Bitwise OR: \|](https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html#operator_bitwise-or)
+
+The `|` operator does a bitwise-or.
+
+For example a bitwise-or of `1010` and `1100` returns `1110` as the first 3 bits are set in either the first or second value but neither value sets the last bit.
+
+```
+1010
+1100
+----
+1110
+```
+
+An the same in SQL:
+
+```sql
+SELECT CONV(b'1010' | b'1100',10,2);
+```
+
+```
++------------------------------+
+| CONV(b'1010' | b'1100',10,2) |
++------------------------------+
+| 1110                         |
++------------------------------+
+1 row in set (0.00 sec)
+```
+
+## [Bitwise XOR: ^](https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html#operator_bitwise-xor)
+
+The `^` operator does an bitwise-xor (exclusive or).
+
+For example:
+
+```
+1010
+1100
+----
+0110
+```
+
+The same in SQL:
+
+```sql
+SELECT CONV(b'1010' ^ b'1100',10,2);
+```
+
+```
++------------------------------+
+| CONV(b'1010' ^ b'1100',10,2) |
++------------------------------+
+| 110                          |
++------------------------------+
+1 row in set (0.00 sec)
+```
+
+Here the value is shown as `110` instead of `0110` as leading zero's are removed.
+
+## [Left shift: <<](https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html#operator_left-shift)
+
+The `<<` operator does a left shift operation.
+
+In the following example we use `1<<n` to move a value of `1` `n` places to the left.
+
+```sql
+WITH RECURSIVE cte(n) AS (
+    SELECT 0 AS n
+    UNION ALL
+    SELECT 1+n FROM cte WHERE n<10
+)
+SELECT n,1<<n,LPAD(CONV(1<<n,10,2),11,0) FROM cte;
+```
+
+```
++------+------+----------------------------+
+| n    | 1<<n | LPAD(CONV(1<<n,10,2),11,0) |
++------+------+----------------------------+
+|    0 |    1 | 00000000001                |
+|    1 |    2 | 00000000010                |
+|    2 |    4 | 00000000100                |
+|    3 |    8 | 00000001000                |
+|    4 |   16 | 00000010000                |
+|    5 |   32 | 00000100000                |
+|    6 |   64 | 00001000000                |
+|    7 |  128 | 00010000000                |
+|    8 |  256 | 00100000000                |
+|    9 |  512 | 01000000000                |
+|   10 | 1024 | 10000000000                |
++------+------+----------------------------+
+11 rows in set (0.00 sec)
+```
+
+## [Right shift: >>](https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html#operator_right-shift)
+
+The `>>` operator does a right shift operation.
+
+In the following example we use `1024>>n` to move a value of `1024` (`10000000000` in binary) `n` places to the right.
+
+```sql
+WITH RECURSIVE cte(n) AS (
+    SELECT 0 AS n
+    UNION ALL
+    SELECT n+1 FROM cte WHERE n<10
```

Comment:
Maybe include a shift of 11 to show that no warnings etc comes from shifting `1`'s out?
```suggestion
    SELECT n+1 FROM cte WHERE n<11
```